stac_api/api/extensions/tiles_search.py

```diff
@@ -0,0 +1,231 @@
+from rio_tiler.constants import WEB_MERCATOR_TMS
+from rio_tiler.io import BaseReader, STACReader
+from rio_tiler.models import ImageData
+from rio_tiler.mosaic import mosaic_reader
+from titiler.dependencies import DefaultDependency
+from titiler.endpoints.factory import MosaicTilerFactory
+
+from stac_api.api.extensions.extension import ApiExtension
+from stac_api.clients.postgres.core import CoreCrudClient
+
+
+@attr.s
+class FakeRequest:
+    base_url: str = "http://localhost:8081"
+    query_params: Dict = {}
+
+
+@attr.s
+class DynamicStacBackend(BaseBackend):
+    """Like a STAC backend but dynamic"""
+
+    path: str = attr.ib()
+    reader: Type[BaseReader] = attr.ib(default=STACReader)
+    reader_options: Dict = attr.ib(factory=dict)
+    backend_options: Dict = attr.ib(factory=dict)
+
+    item_collection: Dict = attr.ib(factory=dict)
+
+    # default values for bounds and zoom
+    bounds: Tuple[float, float, float, float] = attr.ib(default=(-180, -90, 180, 90))
+    minzoom: int = attr.ib(default=0)
+    maxzoom: int = attr.ib(default=30)
+    mosaic_quadkey_zoom: Optional[int] = attr.ib(default=None)
+
+    # Because we are not using mosaicjson we are not limited to the WebMercator TMS
+    tms: TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
+
+    mosaic_def: MosaicJSON = attr.ib(init=False)
+
+    _backend_name = "DynamicSTAC"
+
+    def __attrs_post_init__(self):
+        """Post Init."""
+        # Construct a FAKE mosaicJSON
+        self.mosaic_def = MosaicJSON(
+            mosaicjson="0.0.2",
+            name="it's fake but it's ok",
+            minzoom=self.minzoom,
+            maxzoom=self.maxzoom,
+            quadkey_zoom=self.mosaic_quadkey_zoom,
+            tiles=[],
+        )
+
+    def write(self, overwrite: bool = True):
+        """Write mosaicjson document."""
+        pass
+
+    def update(self):
+        pass
+
+    def _read(self) -> MosaicJSON:
+        pass
+
+    def assets_for_tile(self, x: int, y: int, z: int, **kwargs) -> List[str]:
+        """Retrieve assets for tile."""
+        bounds = self.tms.bounds(x, y, z)
+        return self.get_assets(list(bounds), **kwargs)
+
+    def assets_for_point(self, lng: float, lat: float) -> List[str]:
+        """Retrieve assets for point."""
+        # geom = {"type": "Point", "coordinates": [lng, lat]}
+        bounds = [lng, lat, lng, lat]
+        return self.get_assets(bounds)
+
+    def get_assets(self, bbox, **kwargs) -> List[str]:
+        """Find assets."""
+        print("get assets kwargs")
+        print(kwargs)
+        kwargs["bbox"] = bbox
+        client = self.reader_options["client"]
+
+        kwargs["request"] = FakeRequest()
+
+        print("get_assets edited kwargs")
+        print(kwargs)
+
+        feature_collection = client.get_search(**kwargs)
+
+        print("feature_collection")
+        print(feature_collection)
+        return [default_stac_accessor(f) for f in feature_collection["features"]]
```

Comment:
I don't fully understand this... seems like here I'm returning the _id_ of each stac item, rather than the stac items themselves... Seems like it would be better to return `feature_collection['features']` and then use the resulting STAC dicts directly.